### DISCUSSION TITLE
INA260

### DISCUSSION THREAD

**Discussion by Blu-Eagle** [arendst/Tasmota#11754] [Show and tell]
Hi,
I have modified (quickly) INA219 so that now I an able to use INA260.
it tells INA219, but interfaces INA260.
![tasmota](https://user-images.githubusercontent.com/32535727/115063877-75c13980-9eec-11eb-9e86-abb320ed06ad.png)
I am not skilled enough to add it to tasmota, can anyone help me? you can find attached the ina-file I created (compressed)?
of course is quick and dirty, but it works..... if you want to try it.

In case you wonder why: I am logging 2x100W solar panels (Lxx, temperature, hum, voltage, current, and power) everything is sent via MQTT to node-red and to Influx DB. I use Grafana to review the data. Hum & temp with 2 sensors and the Analog input is used to check Ina voltage (I use a calibrated resistive partitor to reduce the solar panel input for 8266).

[xsns_13_ina219.ino.zip](https://github.com/arendst/Tasmota/files/6327072/xsns_13_ina219.ino.zip)

thank you
giuseppe

**Comment by barbudor** [comment]
What advantage have you seen in the IN260 vs the already supported IN219 and INA226 ?
A part from the integrated shunt, I can't see an obvious advantage of the INA260, especially comparing to the INA226
The lack of ready module on AliExpress do not motivate me.

**Reply by Blu-Eagle** [reply]
INA260 works up to 15A, while INA219 works up to 3.2A.

**Reply by barbudor** [reply]
Wrong
INA219 has the flexibility to work with whatever shunt resistor you use
See examples in https://tasmota.github.io/docs/Commands/#sensor13
For example 32A with a 10mohms shunt

**Reply by Blu-Eagle** [reply]
I know, but I got the ina260 from amazon in 1 day, while it is not easy to find shunt resistors of 10mOhm... unless you buy them from Cina and get to Italy in 1 month via normal mail.
anyway it would be nice to add Ina260 to Tasmota library, can you help?

**Reply by saper-2** [reply]
You mention Italy, check those electronic warehouse: arrow, mouser, farnell and I see TME is also in Italy too - but you should know some of them since you playing with electronic already 😄 ...

Like @barbudor mentioned, INA219 is more "universal" and more common available (in module format too). If you want add driver, I think the best would be to write one, or figure out how to nicely integrate INA260 into INA219 driver and go for pull request - write in driver comments and/or in PR what and how you did - it'll be easier to revise/adopt it by someone and it's likely to show up in some future version as built-in (if there is not much changes/additional code) or "for custom build". 

And you don't need to look in china for parts (unless you like wasting some money for counterfeit/old or badly stored in stock that results in them to being annoying to solder) e.g.: 4-terminal shunt resistor 0,5% 10mOhm 3W (that's about ~17A it can handle given proper cooling): https://eu.mouser.com/ProductDetail/Vishay-Dale/WSL3637R0100DEA?qs=5qr1tyFj3NLzr0u0T49OJA%3D%3D 

I did a quick calc: ina219 (sot-23-8) + shunt resistor (3W/0.5%/50ppm/0.01&Omega;) < 4€ , INA260 > 4,5€ - math is brutal 😅 , when I pick up parts (and design the circuit) I look at final price already to be optimized (lowest).
